### PR TITLE
build: build with SDK min of 10.14

### LINKED ISSUE #19442: Bump macOS SDK to 10.14
Refs: https://github.com/electron/electron/pull/19166/files#diff-e11aa9f97cd49dc0e190bbe776261664R24-R26

Once crbug.com/986701 is fixed we should bump our SDK requirement is 10.14, this won't affect the minimum supported macOS *runtime* version.

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Fixes #19442.

As [crbug.com/986701](https://crbug.com/986701) has been closed, this should be safe to enable once more.

cc @deepak1556 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Upgraded macOS SDK to 10.14 Mojave.

### CODE REVIEW

File: build/args/all.gn [hunk 1]
```diff
@@ -21,6 +21,4 @@ angle_enable_vulkan_validation_layers = false
 
 is_cfi = false
 
-# TODO: Remove this and update CI to contain 10.14 SDK once
-# crbug.com/986701 is fixed.
-mac_sdk_min = "10.13"
+mac_sdk_min = "10.14"
```
Comment (deepak1556) [inline build/args/all.gn]: This arg can be removed, its set to `10.14` in upstream, we were just working around it.

**Comment by miniak** [comment]
@codebytere #19918 can be merged after this one

**Comment by miniak** [comment]
Fixes #19442

**Comment by MarshallOfSound** [comment]
Consistent macOS failure on `10.14`.  Kool

**Comment by zcbenz** [comment]
Can you mention this in release notes? Changing version of SDK should cause cautions from users as it affects behaviors of apps.

**Comment by codebytere** [comment]
Closing for now; this failure is deterministic but impossible to recreate locally. I'll come back to it later.

**Pull request closed without merging**